components/nodemanager-service/pgdb/filtering.go

```diff
@@ -92,8 +91,10 @@ func buildWhereFilter(mergeableFilters []*common.Filter, tableAbbrev string, fil
 
 // Builds an IN where condition like: j.parent_id IN ('e57605ed-bb8a-49b8-606c-af0e2b31b139')
 func whereFieldIn(field string, arr []string, tableAbbrev string) (condition string, err error) {
-	if !utils.IsSqlSafe(field) {
-		return "", &utils.InvalidError{Msg: fmt.Sprintf("Unsupported character found in field: %s", field)}
+	if !acceptedJSONFilterFields(field) {
+		if !utils.IsSqlSafe(field) {
+			return "", &utils.InvalidError{Msg: fmt.Sprintf("Unsupported character found in field: %s", field)}
+		}
```

Comment:
I know this is not all new code from this PR, but it's not clear what this is guarding against. All of the possible values this function could receive are hardcoded in the `nodesFilterField` map in `pgdb/nodes.go`. If we still aren't comfortable interpolating plain strings into our SQL (which I think is totally reasonable!), `pq` has a function that I think should guard against any shenanigans here: https://godoc.org/github.com/lib/pq#QuoteIdentifier

Reply:
> All of the possible values this function could receive are hardcoded 

Are they? tags aren't hardcoded, iirc. those just get passed through

Reply:
https://github.com/chef/automate/blob/master/components/nodemanager-service/pgdb/nodes.go#L582
https://github.com/chef/automate/blob/master/components/nodemanager-service/pgdb/filtering.go#L57

Reply:
though i guess that means we need the issqlsafe bit there but elsewhere? hmmm. im not sure.

Reply:
Ahh yeah, I forgot about the tags.